editor/src/utils/deep-equality.ts

```diff
@@ -809,6 +809,116 @@ export function combine14EqualityCalls<A, B, C, D, E, F, G, H, I, J, K, L, M, N,
   }
 }
 
+export function combine15EqualityCalls<A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, X>(
```

Comment:
maybe we could even generate a bespoke equality function for each type we want to support. We have some experience with the TS AST API so I think this would be an interesting thing to try